drivers/sensor/sgp40/sgp40.c

```diff
@@ -0,0 +1,310 @@
+/*
+ * Copyright (c) 2021 Leonard Pollak
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT sensirion_sgp40
+
+#include <device.h>
+#include <drivers/i2c.h>
+#include <kernel.h>
+#include <drivers/sensor.h>
+#include <sys/__assert.h>
+#include <logging/log.h>
+#include <sys/byteorder.h>
+#include <sys/crc.h>
+
+#include <drivers/sensor/sgp40.h>
+#include "sgp40.h"
+
+LOG_MODULE_REGISTER(SGP40, CONFIG_SENSOR_LOG_LEVEL);
+
+static uint8_t sgp40_compute_crc(uint16_t value)
+{
+	uint8_t buf[2];
+	sys_put_be16(value, buf);
+
+	return crc8(buf, 2, SGP40_CRC_POLY, SGP40_CRC_INIT, false);
+}
+
+static int sgp40_write_command(const struct device *dev, uint16_t cmd)
+{
+
+	const struct sgp40_config *cfg = dev->config;
+	uint8_t tx_buf[2];
+	sys_put_be16(cmd, tx_buf);
+
+	return i2c_write(cfg->bus, tx_buf, sizeof(tx_buf),
+			 cfg->i2c_addr);
+}
+
+static int sgp40_start_measurement(const struct device *dev)
+{
+	const struct sgp40_config *cfg = dev->config;
+	struct sgp40_data *data = dev->data;
+
+	uint8_t tx_buf[8] = { SGP40_CMD_MEASURE_RAW >> 8 ,
+		SGP40_CMD_MEASURE_RAW & 0xFF,
```

Comment:
byteorder API can also be used here

Reply:
updated. i am not really sure about this one